File: sdk/python/examples/first_run.py
"""Connect locally, run one turn, and close the SDK connection."""

import argparse
import asyncio
from pathlib import Path

from dotcraft import DotCraft, LocalOptions


async def main() -> None:
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument("workspace")
    args = parser.parse_args()

    options = LocalOptions(workspace_path=str(Path(args.workspace).resolve()))
    async with await DotCraft.connect_local(options) as dotcraft:
        thread = await dotcraft.threads.start(user_id="sdk-example")
        result = await thread.run("Summarize this workspace in three bullets.")
        print(result.text)


if __name__ == "__main__":
    asyncio.run(main())

File: sdk/typescript/samples/applications/first-run.ts
import { DotCraft } from "@dotcraft/sdk";

const workspacePath = process.argv[2];
if (!workspacePath) throw new Error("Usage: first-run <workspace-path>");

const dotcraft = await DotCraft.local({ workspacePath });
try {
  const thread = await dotcraft.threads.start({ userId: "sdk-example" });
  const result = await thread.run("Summarize this workspace in three bullets.");
  console.log(result.text);
} finally {
  await dotcraft.close();
}
